rustls/src/server/tls12.rs

```diff
@@ -208,11 +214,15 @@ mod client_hello {
                 server_key.get_key(),
                 &self.randoms,
             )?;
-            let doing_client_auth = emit_certificate_req(&mut self.transcript, cx)?;
+            let doing_client_auth = emit_certificate_req(&self.config, &mut self.transcript, cx)?;
             emit_server_hello_done(&mut self.transcript, &mut cx.common);
 
             if doing_client_auth {
                 Ok(Box::new(ExpectCertificate {
+                    key_log: self.config.key_log.clone(),
+                    verifier: self.config.verifier.clone(),
+                    ticketer: self.config.ticketer.clone(),
+                    session_storage: self.config.session_storage.clone(),
```

Comment:
I think to make progress here, we should just keep threading the whole `config` instead of breaking it apart. If we do that then we'll be able to eliminate all the extra `clone()`s introduced and still be able to accomplish the main goal.